### PR TITLE
Add command to actually drop all tables and re-run migration

### PR DESCRIPTION
**Pull request by notbakaneko**
`migrate:fresh` only drops the tables on a single connection but still runs the migrations for every connection which makes no sense.

Laravel has no plans to address this, so here we are.

Also sets up the schemas for the indices.

### CODE REVIEW

**Comment by peppy** [comment]
Doesn't seem to be working here:

image

**Comment by notbakaneko** [comment]
it's `migrate:fresh-all`

**Comment by peppy** [comment]
oh.

**Comment by notbakaneko** [comment]
changed it to override the existing `migrate:fresh` command to be probably less confusing.

**Comment by peppy** [comment]
Seems to run as expected.

```
Seeding: MiscSeeder
Seeding: CountrySeeder
Seeding: UserSeeder
Seeding: BeatmapSeeder
Seeding Beatmaps, this may take a while...
Error: No OSU_API_KEY value set in .env file. Can't seed beatmap data!
Seeding: EventSeeder
Can't seed events due to having no beatmap data.
Seeding: ScoreSeeder
Can't seed Scores due to having no beatmap data.
Seeding: BanchoStatsSeeder
Seeding: ForumSeeder
{"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"failed to parse [type]"}],"type":"mapper_parsing_exception","reason":"failed to parse [type]","caused_by":{"type":"illegal_state_exception","reason":"Can't get text on a START_OBJECT at 1:734"}},"status":400}
Seeding: UserProfileSeeder
Can't seed favourite maps, map playcounts or leaders due to having no beatmap data.
Seeding: ProductSeeder
{"error":{"root_cause":[{"type":"mapper_parsing_exception","reason":"failed to parse [type]"}],"type":"mapper_parsing_exception","reason":"failed to parse [type]","caused_by":{"type":"illegal_state_exception","reason":"Can't get text on a START_OBJECT at 1:734"}},"status":400}```

Are the two errors here expected when `OSU_API_KEY` is missing, or unrelated failures?

File: app/Providers/MigrationServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace App\Providers;
+
+use App\Console\Commands\MigrateFreshAllCommand;
+use Illuminate\Database\MigrationServiceProvider as BaseMigrationServiceProvider;
```
Comment (nanaya) [inline app/Providers/MigrationServiceProvider.php]: but it's not this provider which registers `command.migrate.fresh`...? :thinking:

**Pull request merged**